modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/config/ExtensionConfigNamespaceHandler.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.config;
+
+import org.mule.config.spring.handlers.AbstractMuleNamespaceHandler;
+import org.mule.config.spring.parsers.generic.ChildDefinitionParser;
+import org.mule.config.spring.parsers.specific.TimeFactoryBean;
+
+import org.springframework.beans.factory.xml.NamespaceHandler;
+
+/**
+ * A {@link NamespaceHandler} for elements that configure how the platform
+ * handles extensions
+ *
+ * @since 4.0
+ */
+public class ExtensionConfigNamespaceHandler extends AbstractMuleNamespaceHandler
```

Comment:
ExtensionNamespaceHandler is the generic and dynamic one which can handle any extension. There's an instance of it per each discovered extension. Each instance is bound to each extension's namespace

This one is unique and is bounded uniquely to the 'extension' namespace. This one is static and serves one purpose only which is configuring platform wide extensions properties.

It makes sense to have them separate because they have different concerns and purposes